### PR TITLE
Integration with Microsoft Bot Framework

### PR DESCRIPTION
**Pull request by vkosuri**
Closes https://github.com/gunthercox/ChatterBot/issues/290

### CODE REVIEW

File: chatterbot/adapters/output/microsoft.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
+from chatterbot.adapters.output import OutputAdapter
+import requests
+import json
+
+
+class Microsoft(OutputAdapter):
+    """
+    An output adapter that allows a ChatterBot instance to send
+    responses to a HipChat room.
```
Comment (vkosuri) [inline chatterbot/adapters/output/microsoft.py]: Copy paste error

**Comment by gunthercox** [comment]
@vkosuri Looks good, is this ready to be merged?

**Comment by vkosuri** [comment]
Thanks , I am working on documentation part also, i will push some more
comits. by today.

**Comment by vkosuri** [comment]
@gunthercox i haven't added tests for this PR, i will add soon, do you have any comments/suggestions on this PR?

**Comment by gunthercox** [comment]
@vkosuri From what I can see, this looks fantastic.

In the examples, is the value for `directline_access_token_or_secret` an actual access token?

**Comment by vkosuri** [comment]
Yes, https://docs.botframework.com/en-us/restapi/directline/

**Comment by vkosuri** [comment]
@gunthercox any comments/suggestions?

**Comment by gunthercox** [comment]
**Comments & feedback**

- Overall, the code looks fantastic.
- I'm very glad to see that logging statements were included.
- It might be useful to include a simple example program in the `examples` directory. This would also be helpful for manual testing since there aren't any unit tests yet.

I would be happy to help write tests for this, let me know.

**Comment by vkosuri** [comment]
- [ ] Input adapter
- [ ] Output adapter
- [ ] Tests
- [ ] Documentation
- [ ] Example Program
- [ ] Logging

**Comment by vkosuri** [comment]
@gunthercox any comments/suggestions?

**Comment by gunthercox** [comment]
How did you get the value to use for the `conversation_id`? I'm trying to set up the example to run locally, do I need to manually authenticate with the API and start a new conversation to get a conversation_id?

**Comment by vkosuri** [comment]
You eill get it from here

```
Request URL
https://directline.botframework.com/api/conversations
Response Body
{
  "conversationId": "ENwpIpwOny6",
  "token": "xtFDtPemROU.dAA.RQBOAHcAcABJAHAAdwBPAG4AeQA2AA.IEz11QI90gE.XXui3MZv4-8.kg2KPXCgTU8mcEE_DB07mUQHKG_C-t6c-ISlI0UDM0Q",
  "expires_in": 0
}
```

**Comment by gunthercox** [comment]
I believe all the changes here look good.

**Pull request merged**